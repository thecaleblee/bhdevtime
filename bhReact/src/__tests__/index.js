import React from "react"
import {render} from 'react-testing-library'
import renderer from 'react-test-renderer'
import { StaticQuery } from "gatsby"
import Index from "../pages/index"
import {toBeInTheDocument} from 'jest-dom'
expect.extend({toBeInTheDocument})

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `BHFOSDEV Caleb Masters`,
        },
      },
    })
  )
})

describe(`Index`, () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Index />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`contains a greating`, () => {
    const { getByText } = render(<Index />)
    const greeting = getByText(`Purpose`)
    expect(greeting).toBeInTheDocument()
  })
})