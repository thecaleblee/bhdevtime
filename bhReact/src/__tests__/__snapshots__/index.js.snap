// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<div>
  <div
    className="sc-bdVaJa dSFqaK"
  >
    <a
      className="btn-lg"
      href="/"
    >
      Home
    </a>
    <a
      className="btn-lg"
      href="/wordpress-hosting"
    >
      Wordpress-Hosting
    </a>
    <a
      className="btn-lg"
      href="/"
    >
      Reload Site
    </a>
  </div>
  <h1
    className="sc-htpNat cRYqJk"
  >
    Purpose
  </h1>
  <p
    className="sc-bxivhb fXVBn"
  >
    To test the ability to use React and Contentful to create a static version of BH.components using components, with test coverage, while allowing editing from non-developers via Contentful.
  </p>
  <footer
    className="sc-bwzfXH fQNGhL"
  >
    Â© 
    2019
     BHFOSDEV, Bluehost, calebmasters
  </footer>
</div>
`;
